Clarify Semantics breaks Slicing on vectors with shape 1

To reproduce:

```
	v = tensor.New(
		tensor.WithShape(1),
		tensor.WithBacking([]float64{2}),
	)
	sliced, err = v.Slice(gorgonia.S(0))
	handleErr(err)
	fmt.Printf("[0:1]\n%v\n%v\n\n", sliced.Shape(), sliced)
```
yields

```
panic: runtime error: index out of range [0] with length 0
goroutine 1 [running]:
gorgonia.org/tensor.(*AP).S(0xc000298400, 0x1, 0xc000159e98, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/dcuadrado/go/src/gorgonia.org/tensor/ap.go:243 +0xcfc
```

h/t @dcu

--- a/dense_matop_test.go
+++ b/dense_matop_test.go
@@ -531,6 +531,11 @@ var denseSliceTests = []struct {
 	correctStride []int
 	correctData   interface{}
 }{
+	// scalar-equiv vector (issue 102)
+	{"a[0], a is scalar-equiv", []float64{2},
+		Shape{1}, []Slice{ss(0)}, ScalarShape(), nil, 2.0},
+
+	// vector
 	{"a[0]", []bool{true, true, false, false, false},
 		Shape{5}, []Slice{ss(0)}, ScalarShape(), nil, true},
 	{"a[0:2]", Range(Byte, 0, 5), Shape{5}, []Slice{makeRS(0, 2)}, Shape{2}, []int{1}, []byte{0, 1}},
@@ -632,7 +637,6 @@ func TestDense_Slice(t *testing.T) {
 	if err == nil {
 		t.Error("Expected a IndexError")
 	}
-
 }
 
 func TestDense_SliceInto(t *testing.T) {

--- a/shape.go
+++ b/shape.go
@@ -26,7 +26,7 @@ func (s Shape) TotalSize() int {
 
 // CalcStrides calculates the default strides for a shape
 func (s Shape) CalcStrides() []int {
-	if s.IsScalarEquiv() {
+	if s.IsScalar() {
 		return nil
 	}
 

--- a/shape_test.go
+++ b/shape_test.go
@@ -52,23 +52,26 @@ func TestShapeIsX(t *testing.T) {
 	assert.False(s.IsColVec())
 	assert.False(s.IsRowVec())
 
+	// vectors
+
+	// scalar-equiv vector
 	s = Shape{1}
 	assert.False(s.IsScalar())
 	assert.True(s.IsScalarEquiv())
 	assert.True(s.IsVector())
 	assert.True(s.IsVectorLike())
 	assert.True(s.IsVector())
-
 	assert.False(s.IsColVec())
 	assert.False(s.IsRowVec())
 
-	// vector
+	// vanila vector
 	s = Shape{2}
 	assert.False(s.IsScalar())
 	assert.True(s.IsVector())
 	assert.False(s.IsColVec())
 	assert.False(s.IsRowVec())
 
+	// col vec
 	s = Shape{2, 1}
 	assert.False(s.IsScalar())
 	assert.True(s.IsVector())
@@ -76,6 +79,7 @@ func TestShapeIsX(t *testing.T) {
 	assert.True(s.IsColVec())
 	assert.False(s.IsRowVec())
 
+	// row vec
 	s = Shape{1, 2}
 	assert.False(s.IsScalar())
 	assert.True(s.IsVector())
@@ -90,6 +94,7 @@ func TestShapeIsX(t *testing.T) {
 	assert.False(s.IsColVec())
 	assert.False(s.IsRowVec())
 
+	// scalar equiv matrix
 	s = Shape{1, 1}
 	assert.False(s.IsScalar())
 	assert.True(s.IsScalarEquiv())
@@ -105,10 +110,10 @@ func TestShapeCalcStride(t *testing.T) {
 	s = Shape{}
 	assert.Nil(s.CalcStrides())
 
-	s = Shape{1}
-	assert.Nil(s.CalcStrides())
-
 	// vector shape
+	s = Shape{1}
+	assert.Equal([]int{1}, s.CalcStrides())
+
 	s = Shape{2, 1}
 	assert.Equal([]int{1, 1}, s.CalcStrides())
 

